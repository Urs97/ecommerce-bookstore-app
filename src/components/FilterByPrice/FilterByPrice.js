import React, { useState, useEffect, useRef, useContext } from 'react';
import "nouislider/distribute/nouislider.css";
import './FilterByPrice.scss';

import StoreContext from '../../context/StoreContext';
import Nouislider from "nouislider-react";

const FilterByPrice = () => {

    const context = useContext(StoreContext);
    
    const [minPrice, setMinPrice] = useState(0);
    const [maxPrice, setMaxPrice] = useState(0);
    const filterRef = useRef(null);
    const data = context.data;

    useEffect(() => {
        if (minPrice !== 0 || maxPrice !== 100) {
            if (filterRef.current && filterRef.current.noUiSlider) {
                filterRef.current.noUiSlider.reset();
            }
        }
    // eslint-disable-next-line
    }, [context.state.filterSliderState])

    const onChange = () => (render, handle, value, un, percent) => {
        setMinPrice(value[0]);
        setMaxPrice(value[1]);
    }

    return (
        <section className="filter-by-price">
            <h4 className="sidebar-title">Filter By Price</h4>
            <Nouislider 
                instanceRef={instance => filterRef.current = instance }
                range={{ min: 0, max: 100 }} 
                start={[0, 100]}
                step={10} 
                connect 
                onUpdate={onChange()}/>
            <div className="filter-by-price-div">
                <button 
                    className="button" 
                    onClick={() => context.submitFilterValue(minPrice, maxPrice, data)}
                >
                    Filter
                </button>
                <span>${minPrice} - ${maxPrice}</span>
            </div>
        </section>
    )
};

export default FilterByPrice;