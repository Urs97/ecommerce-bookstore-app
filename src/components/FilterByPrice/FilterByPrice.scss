@import "../../styles/shared.scss";

.filter-by-price {
  padding: $vertical-sidebar-padding 0;
  border-bottom: $divider;

  & input {
    display: block;
    width: 100%;
  }

  & .filter-by-price-div {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding-bottom: 0.5em;

    & button {
      border: none;
      padding: 0.6em 1em;
    }

    & span {
      color: $text-color;
      font-size: 1.25em;
    }
  }
}

.filter-by-price > *:not(:last-child) {
  margin-bottom: $vertical-sidebar-padding;
}

// NoUiSlider styling

.noUi-target {
  background-color: #cccccc;
  border: 1px solid #cccccc;
  border-radius: 1em;
  box-shadow: inset 0 1px 1px #cccccc, 0 3px 6px -5px #cccccc;
}

.noUi-horizontal {
  height: 0.75em;
}

.noUi-horizontal .noUi-handle {
  width: 1.5em;
  height: 1.5em;
}

.noUi-handle {
  background-color: $dark-blue;
  border-radius: 1em;
  border: 1px solid $dark-blue;
  box-shadow: none;
}

.noUi-handle:before,
.noUi-handle:after {
  background: none;
}

.noUi-touch-area {
  background-color: $dark-blue;
  border-radius: 1em;
  border-color: $dark-blue;
}

.noUi-connect {
  background-color: $dark-blue;
}
