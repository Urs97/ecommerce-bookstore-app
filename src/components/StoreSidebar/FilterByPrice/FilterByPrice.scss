@import '../../../styles/shared.scss';

.filter-by-price {
    padding: $vertical-sidebar-padding 0;
    border-bottom: $sidebar-divider;

    & input {
        display: block;
        width: 100%;
    }

    & .filter-by-price-div {
        display: flex;
        align-items: center;
        justify-content: space-between;
        padding-bottom: 0.5em;

        & button {
            border: none;
            padding: 0.6em 1em;
        }
    }
}

.filter-by-price > *:not(:last-child) {
    margin-bottom: $vertical-sidebar-padding;
}

// NoUiSlider styling

.noUi-target {
    background-color: #CCCCCC;
    border: 1px solid #CCCCCC;
    border-radius: 1em;
    box-shadow: inset 0 1px 1px #CCCCCC, 0 3px 6px -5px #CCCCCC;
}

.noUi-horizontal {
    height: 0.75em;
}

.noUi-horizontal .noUi-handle {
    width: 1.5em;
    height: 1.5em;
}

.noUi-handle {
    background-color: $dark-blue;
    border-radius: 1em;
    border: 1px solid $dark-blue;
    box-shadow: none;
}

.noUi-handle:before, .noUi-handle:after {
    background: none;
}

.noUi-touch-area {
    background-color: $dark-blue;
    border-radius: 1em;
    border-color: $dark-blue;
}

.noUi-connect {
    background-color: $dark-blue;
}