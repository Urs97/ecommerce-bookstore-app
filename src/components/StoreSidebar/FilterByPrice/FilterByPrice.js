import React, { useState } from 'react';
import Nouislider from "nouislider-react";
import "nouislider/distribute/nouislider.css";
import './FilterByPrice.scss';

function FilterByPrice() {
    const [minPrice, setMinPrice] = useState(0);
    const [maxPrice, setMaxPrice] = useState(0);

    const onChange = () => (render, handle, value, un, percent) => {
        setMinPrice(value[0]);
        setMaxPrice(value[1]);
    }

    return (
        <section className="filter-by-price">
            <h4 className="sidebar-title">Filter By Price</h4>
            <Nouislider 
                range={{ min: 0, max: 100 }} 
                start={[20, 80]}
                step={10} 
                connect 
                onUpdate={onChange()}/>
            <div className="filter-by-price-div">
                <button className="button">Filter</button>
                <span>{minPrice}-{maxPrice}</span>
            </div>
        </section>
    )
};

export default FilterByPrice;