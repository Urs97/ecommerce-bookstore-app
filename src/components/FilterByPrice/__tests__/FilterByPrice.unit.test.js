import React from 'react';
import {shallow} from 'enzyme';
import FilterByPrice from '../FilterByPrice';

describe('FilterByPrice', () => {
    test('renders FilterByPrice', () => {
        const wrapper = shallow(
            <FilterByPrice />
        );
        expect(wrapper).toMatchSnapshot();
    });
});