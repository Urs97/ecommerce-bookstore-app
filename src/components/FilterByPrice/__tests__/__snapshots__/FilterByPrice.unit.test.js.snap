// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterByPrice renders FilterByPrice 1`] = `
<section
  className="filter-by-price"
>
  <h4
    className="sidebar-title"
  >
    Filter By Price
  </h4>
  <Memo(Nouislider)
    connect={true}
    instanceRef={[Function]}
    onUpdate={[Function]}
    range={
      Object {
        "max": 100,
        "min": 0,
      }
    }
    start={
      Array [
        0,
        100,
      ]
    }
    step={10}
  />
  <div
    className="filter-by-price-div"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      Filter
    </button>
    <span>
      $
      0
       - $
      0
    </span>
  </div>
</section>
`;

exports[`FilterByPrice renders empty FilterByPrice 1`] = `
<section
  className="filter-by-price"
>
  <h4
    className="sidebar-title"
  >
    Filter By Price
  </h4>
  <Memo(Nouislider)
    connect={true}
    instanceRef={[Function]}
    onUpdate={[Function]}
    range={
      Object {
        "max": 100,
        "min": 0,
      }
    }
    start={
      Array [
        0,
        100,
      ]
    }
    step={10}
  />
  <div
    className="filter-by-price-div"
  >
    <button
      className="button"
      onClick={[Function]}
    >
      Filter
    </button>
    <span>
      $
      0
       - $
      0
    </span>
  </div>
</section>
`;
